rc/filetype/elvish.kak

```diff
@@ -22,16 +22,17 @@ provide-module elvish %§
 add-highlighter shared/elvish regions
 add-highlighter shared/elvish/code default-region group
 add-highlighter shared/elvish/double_string region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/elvish/single_string region "'" (?<!')('')*' fill string
+add-highlighter shared/elvish/single_string region "'" ('')*' fill string
 add-highlighter shared/elvish/comment region '#' $ fill comment
 
 add-highlighter shared/elvish/code/variable regex \$[\w\d-_:~]+ 0:variable
 add-highlighter shared/elvish/code/variable_in_assignment regex (?:^|\{\s|\(|\||\;)\h*(?:var|set|tmp|del)((?:\h+[\w\d-_:~]+)*) 1:variable
 add-highlighter shared/elvish/code/variable_in_for regex (?:^|\{\s|\(|\||\;)\h*for\h+([\w\d-_:~]*) 1:variable
 add-highlighter shared/elvish/code/variable_in_except regex \}\h+except\h+([\w\d-_:~]*) 1:variable
 
-add-highlighter shared/elvish/code/builtin regex (?:^|\{\s|\(|\||\;)\h*(var|set|tmp|del|and|or|coalesce|if|while|for|try|fn|pragma|!=|!=s|%|\*|\+|-|-gc|-ifaddrs|-log|-override-wcwidth|-stack|/|<|<=|<=s|<s|==|==s|>|>=|>=s|>s|all|assoc|base|bool|break|call|cd|compare|constantly|continue|count|defer|deprecate|dissoc|drop|each|eawk|echo|eq|eval|exact-num|exec|exit|external|fail|fg|float64|from-json|from-lines|from-terminated|get-env|has-env|has-external|has-key|has-value|is|keys|kind-of|make-map|multi-error|nop|not|not-eq|ns|num|one|only-bytes|only-values|order|peach|pprint|print|printf|put|rand|randint|range|read-line|read-upto|repeat|repr|resolve|return|run-parallel|search-external|set-env|show|sleep|slurp|src|styled|styled-segment|take|tilde-abbr|time|to-json|to-lines|to-string|to-terminated|unset-env|use-mod|wcswidth) 1:builtin
-add-highlighter shared/elvish/code/keyword regex \}\h+(elif|else|except|finally) 1:keyword
+add-highlighter shared/elvish/code/builtin regex (?:^|\{\s|\(|\||\;)\h*(!=|!=s|%|\*|\+|-gc|-ifaddrs|-log|-override-wcwidth|-stack|-|/|<|<=|<=s|<s|==|==s|>|>=|>=s|>s|all|assoc|base|bool|break|call|cd|compare|constantly|continue|count|defer|deprecate|dissoc|drop|each|eawk|echo|eq|eval|exact-num|exec|exit|external|fail|fg|float64|from-json|from-lines|from-terminated|get-env|has-env|has-external|has-key|has-value|is|keys|kind-of|make-map|multi-error|nop|not-eq|not|ns|num|one|only-bytes|only-values|order|peach|pprint|print|printf|put|rand|randint|range|read-line|read-upto|repeat|repr|resolve|return|run-parallel|search-external|set-env|show|sleep|slurp|src|styled|styled-segment|take|tilde-abbr|time|to-json|to-lines|to-string|to-terminated|unset-env|use-mod|wcswidth)(?![-:])\b 1:builtin
+add-highlighter shared/elvish/code/keyword regex (?:^|\{\s|\(|\||\;)\h*(use|var|set|tmp|del|and|or|coalesce|pragma|while|for|try|fn|if)(?![-:])\b 1:keyword
```

Comment:
Is there a non-stylistic reason to put special commands in the `keyword` category rather than the `builtin` category?

If it comes down to stylistic choice, I'd prefer my original categorization of putting special commands in `builtin`:

- Special commands are parsed in the same way as ordinary commands in Elvish.
- `and` (special) and `not` (ordinary) should be in the same category; as should `for` (special) and `each` (ordinary).
- Finally Elvish itself highlight special commands like ordinary commands (for the reasons above).

Reply:
i put them into keyword since they (to me) are part of the language, not "stdlib", that should be consistent with how other languages highlighted.